src/wfs/src/main/java/org/geoserver/wfs/xml/WFSURIHandler.java

```diff
@@ -51,32 +58,95 @@
-            // in order to determine if a request is reflective we need to know what all the 
-            // addresses and hostnames that we are addressable on. Since hostnames are expensive we 
-            // do it once and cache the results
+    
+    static final Set<String> ADDITIONAL_HOSTNAMES = new HashSet<String>();
+    
+    public static void addToParser(GeoServer geoServer, Parser parser) {
+        parser.getURIHandlers().add(0, new WFSURIHandler(geoServer));
+    }
+    
+    // Allows for unit testing.
+    static class InitStrategy {
+        public Collection<NetworkInterface> getNetworkInterfaces(){
             Enumeration<NetworkInterface> e = null;
             try {
                 e = NetworkInterface.getNetworkInterfaces();
             } catch (SocketException ex) {
                 LOGGER.log(Level.WARNING, "Unable to determine network interface info", ex);
             } 
-            while(e != null && e.hasMoreElements()) {
-                NetworkInterface ni = e.nextElement();
-                Enumeration<InetAddress> f = ni.getInetAddresses();
-                while(f.hasMoreElements()) {
-                    InetAddress add = f.nextElement();
+            
+            return Collections.list(e);
+        }
+        
+    }
     
-                    //do the hostname lookup, these are cached after the first call so we only pay the
-                    // price now
-                    add.getHostName();
-                    ADDRESSES.add(add);
+    static {
+        init(new InitStrategy());
+    }
+
+    static void init(InitStrategy strategy) {
+        if (!DISABLED) {
+            initAddresses(strategy);
+            initAliases();
+            initLog();
+        }
+    }
+
+    private static void initLog() {
+        // Log hostnames being treated as reflexive
+        if(LOGGER.isLoggable(Level.INFO)){
+            StringBuilder builder = new StringBuilder("Identified addresses and hostnames for local interfaces: ");
+            boolean first = true;
+            for(InetAddress add: ADDRESSES) {
+                if(!first){
+                    builder.append(", ");
                 }
+                first = false;
+                builder.append(add.getHostAddress());
+                builder.append(" ");
+                builder.append(add.getHostName());
             }
+            LOGGER.log(Level.INFO, builder.toString());
+            
+            builder = new StringBuilder("Additional aliases for local host: ");
+            for(String alias: ADDITIONAL_HOSTNAMES) {
+                builder.append(alias);
+                builder.append(" ");
+            }
+            LOGGER.log(Level.INFO, builder.toString());
         }
     }
 
+    private static void initAliases() {
+        assert ADDITIONAL_HOSTNAMES.isEmpty();
+        // User configurable hostnames
+        String additional=GeoServerExtensions.getProperty(WFSURIHandler.class.getName()+".additionalHostnames");
```

Comment:
Hi, do you have the coding conventions loaded up? This should read "String additional = GeoServerExtensions..."